addons/mail/models/mail_activity.py

```diff
@@ -795,3 +796,130 @@ def activity_unlink(self, act_type_xmlids, user_id=None):
             domain = ['&'] + domain + [('user_id', '=', user_id)]
         self.env['mail.activity'].search(domain).unlink()
         return True
+
+    def _read_progress_bar(self, domain, group_by, progress_bar):
+        if not (progress_bar['field'] == 'activity_state' and self._fields[group_by].store):
+            return super()._read_progress_bar(domain, group_by, progress_bar)
+
+        # use datetime.utcnow() instead of date.today(), because
+        # it's used in _compute_state_from_date and on mocking in tests
+        today_utc = datetime.utcnow()
+        today = date(year=today_utc.year, month=today_utc.month, day=today_utc.day)
+
+        # Code below is a version of _read_group_raw:
+        # * lazy is always False
+        # * extra joins for mail_activity
+        # * extra grouping by mail_activity.res_id
+        # * extra field in select: days_to_deadline
+        # * extra SELECT wrapper to compute and group by activity_state
+        fields = ['activity_state']
+        # we also group by activity_state, but because it's a virtual field, we'll add it later manually
+        groupby = [group_by]
+        offset = 0
+        limit = None
+        orderby = False
+
+        self.check_access_rights('read')
+        query = self._where_calc(domain)
+
+        groupby_list = groupby
+        annotated_groupbys = [self._read_group_process_groupby(gb, query) for gb in groupby_list]
+        groupby_fields = [g['field'] for g in annotated_groupbys]
+        order = orderby or ','.join([g for g in groupby_list])
+        groupby_dict = {gb['groupby']: gb for gb in annotated_groupbys}
+
+        self._apply_ir_rules(query, 'read')
+        aggregated_fields = []
+        select_terms = []
+
+        for gb in annotated_groupbys:
+            select_terms.append('%s as "%s" ' % (gb['qualified_field'], gb['groupby']))
+
+        groupby_terms, orderby_terms = self._read_group_prepare(order, aggregated_fields, annotated_groupbys, query)
+
+        groupby_terms.append('mail_activity.res_id')
+
+        from_clause, where_clause, where_clause_params = query.get_sql()
+        count_field = '_'
+        count_field += '_count'
```

Comment:
I see this is just a column name, IMO there is no need to make it a variable and this could be hardcoded in the query.
```suggestion
        count_field = '__count'
```

Reply:
I wanted to keep close to original read_group_raw code, but ok